src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,37 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, null, 0);
+            if (length == 0)
+            {
+                throw new ArgumentException(SR.Argument_IdnIllegalName, nameof(unicode));
+            }
+
             const int StackAllocThreshold = 512;
             if (count < StackAllocThreshold)
             {
-                char* output = stackalloc char[count];
-                return GetAsciiCore(unicode, count, flags, output, count, reattempt: true);
+                char* output = stackalloc char[length];
+                return GetAsciiCore(unicode, count, flags, output, length);
             }
             else
             {
-                char[] output = new char[count];
+                char[] output = new char[length];
                 fixed (char* pOutput = output)
                 {
-                    return GetAsciiCore(unicode, count, flags, pOutput, count, reattempt: true);
+                    return GetAsciiCore(unicode, count, flags, pOutput, length);
                 }
             }
         }
 
-        private unsafe string GetAsciiCore(char* unicode, int count, uint flags, char* output, int outputLength, bool reattempt)
+        private unsafe string GetAsciiCore(char* unicode, int count, uint flags, char* output, int outputLength)
         {
-            int realLen = Interop.GlobalizationNative.ToAscii(flags, unicode, count, output, outputLength);
-
-            if (realLen == 0)
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, output, outputLength);
+            if (length == 0)
```

Comment:
this check should be now 
 if (length == 0 || length>outputLength)

I understand length should be equal to outputLength but in case something wrong happen for any reason we should account for that